.npmrc

```diff
@@ -0,0 +1,12 @@
+# Hoist these dependencies for workflow tools (linter, typechecker, etc.)
+public-hoist-pattern[]=*prettier*
+public-hoist-pattern[]=@docusaurus/types
+public-hoist-pattern[]=@docusaurus/theme-common
+public-hoist-pattern[]=@types/node
+
+# Hoist these dependencies because development builds in .docusaurus/
+# assume they will be top-level for hot-reloading
+# TODO: re-evaluate whether some of these can become genuine top-level
```

Comment:
I don't understand this comment. -- we want these hoisted, but we may want to have them hoisted?

Reply:
I forwent declaring `react` and `@mdx-js/react` as explicit dependencies because none of our code needs them as a direct dependency. They were, however, expected to be present by `docusaurus start` so they needed to be forcibly hoisted to the top-level `/node_modules`.

The intent is that if/when we ended up making custom React components, `react` the package would be an explicit dependency in `package.json`. This would automatically bring it to the top-level `/node_modules` directory and they would no longer need to be forcibly hoisted (i.e. these lines could be removed).

If they were both not forcibly hoisted and not installed as a dependency, they would live inside pnpm's isolated `node_modules` structure and would be inaccessible from application code (and also `docusaurus start` would error out).

Reply:
Maybe it would be better to just install them as traditional dependencies anyway since they're needed for both `docusaurus start` and `docusaurus build` to work at all. I certainly welcome your opinion.

Reply:
Yeah. If they're required for startup of docusaurus -- I'm actually confused that they are not deps in there, but I guess there's probably some reason for that, which I'm not going to guess to hard about. :)

Let's move them into dependencies in package.json

Reply:
I ended up installing all packages that were otherwise implicitly required for prod/dev tooling to work correctly as actual prod/dev dependencies. All that was left in `.npmrc` was hoisting up `"*prettier*"` (in case any libraries ship with custom prettier plugins), but that string is already one of the default values for `public-hoist-pattern`, so I ended up removing the `.npmrc` file entirely.

Changes made in https://github.com/comcode-org/hackmud_wiki/pull/2/commits/41a76ef480e7c28908dd5e6d9c3bfdca5c107b3d.